From 238493e7f093a64825101b51de037b57a17cf02d Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Fri, 20 Sep 2024 07:15:02 +0800
Subject: [PATCH 042/108] x86-64: Disable PIE on PR gas/32189 test

Disable PIE on PR gas/32189 test, which contains the non-PIE assembly
source, to support GCC defaulted to PIE.

	PR gas/32189
	* testsuite/ld-x86-64/x86-64.exp: Pass $NOPIE_LDFLAGS to linker
	on PR gas/32189 test.

Signed-off-by: H.J. Lu <hjl.tools@gmail.com>
(cherry picked from commit 075c4ca29a1afb1fa19e3038cae5730487e99701)
---
 ld/testsuite/ld-x86-64/x86-64.exp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ld/testsuite/ld-x86-64/x86-64.exp b/ld/testsuite/ld-x86-64/x86-64.exp
index a4f197dcf4d..09b0c04d01c 100644
--- a/ld/testsuite/ld-x86-64/x86-64.exp
+++ b/ld/testsuite/ld-x86-64/x86-64.exp
@@ -1922,7 +1922,7 @@ if { [isnative] && [check_compiler_available] } {
 	] \
 	[list \
 	    "Run pr32189" \
-	    "-nostdlib -nostartfiles" \
+	    "$NOPIE_LDFLAGS -nostdlib -nostartfiles" \
 	    "" \
 	    { pr32189.s } \
 	    "pr32189" \
-- 
2.47.0

